src/yunohost/log.py

```diff
@@ -0,0 +1,424 @@
+    md_path = base_path + METADATA_FILE_EXT
+    if log_path is None:
+        log_path = base_path + LOG_FILE_EXT
+
+    if not os.path.exists(md_path) and not os.path.exists(log_path):
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('log_does_exists', log=path))
+
+    infos = {}
+
+    # If it's a unit operation, display the name and the description
+    if base_path.startswith(CATEGORIES_PATH):
+        infos["description"] = _get_description_from_name(base_filename)
+        infos['name'] = base_filename
+
+    # Display metadata if exist
+    if os.path.exists(md_path):
+        with open(md_path, "r") as md_file:
+            try:
+                metadata = yaml.safe_load(md_file)
+                infos['metadata_path'] = md_path
+                infos['metadata'] = metadata
+                if 'log_path' in metadata:
+                    log_path = metadata['log_path']
+            except yaml.YAMLError:
+                error = m18n.n('log_corrupted_md_file', file=md_path)
+                if os.path.exists(log_path):
+                    logger.warning(error)
+                else:
+                    raise MoulinetteError(errno.EINVAL, error)
+
+    # Display logs if exist
+    if os.path.exists(log_path):
+        from yunohost.service import _tail
+        logs = _tail(log_path, int(number))
+        #logs = [{"datetime": x.split(": ", 1)[0].replace("_", " "), "line": x.split(": ", 1)[1]}  for x in logs if x]
+        infos['log_path'] = log_path
+        infos['logs'] = logs
+
+    return infos
+
+def is_unit_operation(entities='app,domain,service,user', exclude='auth,password', operation_key=None, auto=True):
+    """
+    Configure quickly a unit operation
+
+    This decorator help you to configure quickly the record of a unit operations.
+
+    Argument:
+    entities    A list seperated by coma of entity types related to the unit
+    operation. The entity type is searched inside argument's names of the
+    decorated function. If something match, the argument value is added as
+    related entity.
+
+    exclude     Remove some arguments from the context. By default, arguments
+    called 'password' and 'auth' are removed. If an argument is an object, you
+    need to exclude it or create manually the unit operation without this
+    decorator.
+
+    operation_key   Key describing the unit operation. If you want to display a
+    well formed description you should add a translation key like this
+    "log_" + operation_key in locales files.
+
+    auto        If true, start the recording. If False, the unit operation object
+    created is given to the decorated function as the first argument and you can
+    start recording at the good time.
+    """
+    def decorate(func):
+        def func_wrapper(*args, **kwargs):
+            # For a strange reason we can't use directly the arguments from
+            # is_unit_operation function. We need to store them in a var before.
```

Comment:
You should be able to do here because you don't re-assign the arguments name here so I guess you've runt in this problem before but here you are safe.

This is due to the way python optimize things.

Here is a sample code to demonstrate this:

```python
def pouet(a):
    def prout():
        def pif():
            a + 1
            # uncommenting this will make it fails
            # a = a + 1
            print a
        pif()
    prout()

pouet(42)
```